scipy/stats/tests/test_multivariate.py

```diff
@@ -294,6 +294,24 @@ def _sample_orthonormal_matrix(n):
     u, s, v = scipy.linalg.svd(M)
     return u
 
+def marginal_pdf(X, X_ndim, dimensions, x):
+    # First sort input data based on order of dimensions
+    dimensions = np.asarray(dimensions)
+    dimensions[dimensions < 0] = X_ndim + dimensions[dimensions < 0]
+    dim_sort_idx = dimensions.argsort()
+    x = x[:, dim_sort_idx]
+    
+    i_marginalize = np.ones(X_ndim, dtype=bool)
+    i_marginalize[dimensions] = False
+
+    def g(z):
+        y = np.empty((z.shape[0], x.shape[0], X_ndim))
+        y[..., i_marginalize] = z[:, np.newaxis, :]
+        y[..., ~i_marginalize] = x
+        return X.pdf(y)
+
+    inf = np.full(X_ndim - len(dimensions), np.inf)
+    return cubature(g, -inf, inf).estimate
```

Comment:
this could do with a quick explanation in a comment

Reply:
What do you have in mind? I wrote this to compute reference values from the definition of a marginal distribution.

Reply:
> What do you have in mind?

I don't know that definition by heart, so it would help to informally describe that definition, or otherwise link to it.